pyhf/pdf.py

```diff
@@ -70,68 +88,146 @@ def par_slice(self, name):
     def param_set(self, name):
         return self.par_map[name]['parset']
 
-    def modifier(self, name):
-        return self.par_map[name]['modifier']
+    def modifier_type(self, name):
+        return self.par_map[name]['modifier_type']
 
     def set_poi(self,name):
-        if name not in self.modifiers:
+        if name not in [x for x,_ in self.modifiers]:
             raise exceptions.InvalidModel("The paramter of interest '{0:s}' cannot be fit as it is not declared in the model specification.".format(name))
         s = self.par_slice(name)
         assert s.stop-s.start == 1
         self.poi_index = s.start
 
-    def register_paramset(self, name, n_parameters, modifier):
+    def register_paramset(self, modifier_type, name, n_parameters, parset):
```

Comment:
if indeed `def modifier_type(self, name):` is never used, we might not need the first arg here anymore